modules/dbparser/groupingby.c

```diff
@@ -254,6 +254,30 @@ grouping_by_generate_synthetic_msg(GroupingBy *self, CorrellationContext *contex
   return msg;
 }
 
+static LogMessage *
+grp_sort_remove_context(GroupingBy *self, CorrellationContext *context)
```

Comment:
It seems convenient to group together multiple steps (and I like ), but isn't it against single-responsibility?
I think you just tear up `grouping_by_emit_synthetic` because it contained msg generating and sending it.

Reply:
it kinda does, it has to do with the fact of `CorrellationContext` and `CorrellationState` imho not the best state as of now.
With combining those into this function was to emphasise they should be called together.